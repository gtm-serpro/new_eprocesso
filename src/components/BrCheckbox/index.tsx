import classNames from "classnames";
import uniqueId from "lodash/uniqueId";
import { forwardRef } from "react";

export interface BrCheckboxProps extends React.ComponentPropsWithoutRef<"input"> {
  label?: string;
}

const BrCheckbox = forwardRef<HTMLInputElement, BrCheckboxProps>((props, ref) => {
  const { id = uniqueId("checkbox-"), label, className, ...otherProps } = props;
  const classes = classNames("br-checkbox", !label && "hidden-label", className);

  return (
    <div className={classes}>
      <input id={id} ref={ref} type="checkbox" aria-label={label} {...otherProps} />
      <label htmlFor={id}>{label || id}</label>
    </div>
  );
});

export default BrCheckbox;
