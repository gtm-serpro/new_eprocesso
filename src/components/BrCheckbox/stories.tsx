import { Meta, Story } from "@storybook/react";

import BrCheckbox, { BrCheckboxProps } from ".";

export default {
  title: "Formul√°rios/Checkbox",
  component: BrCheckbox,
} as Meta;

export const SemRotulo: Story<BrCheckboxProps> = (args) => <BrCheckbox {...args} />;

export const ComRotulo: Story<BrCheckboxProps> = (args) => <BrCheckbox {...args} />;
ComRotulo.args = { label: "meu check" };
