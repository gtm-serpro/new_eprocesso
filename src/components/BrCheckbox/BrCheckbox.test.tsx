import { render, screen } from '@testing-library/react';

import BrCheckbox from '.';

describe('<BrCheckbox />', () => {
  it('deveria renderizar apenas um checkbox com o label informado', () => {
    const { container } = render(<BrCheckbox label="meu checkbox" />);

    expect(screen.getAllByRole('checkbox')).toHaveLength(1);
    expect(screen.getByText(/meu checkbox/i)).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  it('deveria renderizar com a classe br-checkbox', () => {
    const { container } = render(<BrCheckbox />);

    expect(container.firstChild).toHaveClass('br-checkbox');
  });

  it('deveria renderizar um checkbox checado', () => {
    render(<BrCheckbox checked onChange={() => ''} />);

    expect(screen.getByRole('checkbox')).toBeChecked();
  });

  it('deveria renderizar um checkbox desabilitado', () => {
    render(<BrCheckbox disabled />);

    expect(screen.getByRole('checkbox')).toBeDisabled();
  });

  it('deveria adicionar id com prefixo se prop nÃ£o informada', () => {
    render(<BrCheckbox label="meu-check" />);
    expect(screen.getByRole('checkbox')).toMatchObject({ id: /^(checkbox-)(\/\w+)+$/ });
  });
});
