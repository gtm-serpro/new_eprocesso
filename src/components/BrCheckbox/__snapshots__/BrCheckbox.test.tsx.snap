// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BrCheckbox /> deveria renderizar apenas um checkbox com o label informado 1`] = `
<div
  class="br-checkbox"
>
  <input
    aria-label="meu checkbox"
    id="checkbox-1"
    type="checkbox"
  />
  <label
    for="checkbox-1"
  >
    meu checkbox
  </label>
</div>
`;
